lib/node-utils/index.ts

```diff
@@ -235,25 +234,26 @@ export function isPromiseHandled(nodeIdentifier: TSESTree.Identifier): boolean {
 }
 
 export function getVariableReferences(
-  context: RuleContext<string, []>,
+  context: TSESLint.RuleContext<string, []>,
   node: TSESTree.Node
 ): TSESLint.Scope.Reference[] {
-  return (
-    (ASTUtils.isVariableDeclarator(node) &&
-      context.getDeclaredVariables(node)[0]?.references?.slice(1)) ||
-    []
-  );
+  if (ASTUtils.isVariableDeclarator(node)) {
+    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+    return context.getDeclaredVariables(node)[0]?.references?.slice(1) ?? [];
+  }
+
+  return [];
```

Comment:
Maybe do an early return instead?

```suggestion
  if (!ASTUtils.isVariableDeclarator(node)) {
    return [];
  }

  // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
  return context.getDeclaredVariables(node)[0]?.references?.slice(1) ?? [];
```

I normally don't like negative conditions, but I'm more in favor of having the "main" thing at the end

Reply:
That would fix it here but could potentially introduce errors in new code, so I've just switched off this rule until there is an option to ignore arrays within conditions.

Reply:
I didn't suggest it because of the disabling of ESLint, but because

> I'm more in favor of having the "main" thing at the end

Reply:
Ah sorry, didn't realize that. I don't see any benefit in this case, I guess it's a matter of taste.